pokemon/src/main/java/gamarket/MoveCollection.java

```diff
@@ -0,0 +1,135 @@
+package gamarket;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Represents all the moves in the Pokemon universe. 
+ * A Pokemon will be able to hold at most 4 moves in its move-set. 
+ */
+public class MoveCollection {
+	
+	/** The current number of MOVEs in the array */
+	private int numMoves; 
+	
+	/** The array to contain all the MOVEs in the Pokemon Universe*/
+	private Move moveArray[]; 
+	
+	/**
+	 * The MoveCollection constructor.
+	 * It sets the numMoves to zero. 
+	 * It allocates a new array of Move type to 7. 
+	 * 
+	 */
+	public MoveCollection()
+	{
+		numMoves = 0; 
+		moveArray = new Move[7];
+	}
+	
+	/**
+	 * The addOrModifyMove fucntion of type void.
+	 * The function adds a new move to the move collection, if the new move does not exist.
+	 * If it does exist, the move information will get updated. 
+	 * @param name This is the Move's name. Should have the Move's name(Tackle). 
+	 * @param type This is the Move's type. Should have the Move's type(Normal). 
+	 * @param category This is the Move's category. Should have the Move's category(Physical).
+	 * @param damage This is the Move's damage. Should have the Move's damage(60). 
+	 */
+	public void addOrModifyMove(String name, String type, String category, String damage)
+	{
+		try {	// running time error prevention.   
+				// double the size of the array if the current array is full. 
+			if(numMoves == moveArray.length-1)
```

Comment:
probably won't ever need to resize the array to something bigger than 4, depending on the implementation you decide to go with

Reply:
I do it like this, because if a data file contains only 4 moves than the array can be a fixed size. But, if the data file contains like 20 moves, than I would need to change the array size again. I coded this way to handle any size of moves that the data file contains so I won't have to keep updating the array size every time in order to store all the moves from the data file.

Reply:
you could also have a flag at top of file that tells how many moves there are